### ISSUE TITLE
Reproduction issues of the "Unified Settings" experiment

### ISSUE THREAD

**Issue by zhujiayi18161778290** [Secure-and-Intelligent-Systems-Lab/WPMixer#10]
Dear author, 
I ran your experiment using all the parameters you set exactly as described, but the results are significantly different from what you presented in your paper. Could you please explain why?
bash ./scripts/Unified/Weather_Unified_setup.sh

Args in experiment: Namespace(model='WPMixer', task_name='long_term_forecast', data='Weather', use_hyperParam_optim=False, no_decomposition=False, use_multi_gpu=False, n_jobs=1, seed=42, seq_len=96, pred_len=96, d_model=128, tfactor=3, dfactor=5, wavelet='sym3', level=1, patch_len=16, stride=8, batch_size=128, learning_rate=0.006434312, dropout=0.2, embedding_dropout=0.0, weight_decay=0.0, patience=5, train_epochs=20, label_len=0, seasonal_patterns='Monthly', features='M', target='OT', freq='h', checkpoints='./checkpoints/', cols=None, num_workers=0, itr=1, lradj='type3', use_amp=True, use_gpu=True, gpu=0, devices='0,1', embed=0, loss='smoothL1', pct_start=0.2, optuna_seq_len=None, optuna_lr=None, optuna_batch=None, optuna_wavelet=None, optuna_tfactor=None, optuna_dfactor=None, optuna_epochs=None, optuna_dropout=None, optuna_embedding_dropout=None, optuna_patch_len=None, optuna_stride=None, optuna_lradj=None, optuna_dmodel=None, optuna_weight_decay=None, optuna_patience=None, optuna_level=None, optuna_trial_num=None, data_path='weather.csv', root_path='./data/weather/', c_in=21, c_out=21, detail_freq='h')
Use GPU: cuda:0
Start Training- WPMixer_Weather_dec-True_sl96_pl96_dm128_bt128_wvsym3_tf3_df5_ptl16_stl8_sd42
train 36696
val 5175
test 10444
Epoch 1: cost time: 41.55 sec
	Epoch 1: Steps- 286 | Train Loss: 0.12689 Vali.MSE: 0.40167 Vali.MAE: 0.27170 Test.MSE: 0.16451 Test.MAE: 0.20738
	Validation loss decreased (inf --> 0.401667).  Saving model ...
Updating learning rate to 0.006434312
Epoch 2: cost time: 49.55 sec
	Epoch 2: Steps- 286 | Train Loss: 0.11572 Vali.MSE: 0.40477 Vali.MAE: 0.26696 Test.MSE: 0.16455 Test.MAE: 0.20480
	EarlyStopping counter: 1 out of 5
Updating learning rate to 0.006434312
Epoch 3: cost time: 50.24 sec
	Epoch 3: Steps- 286 | Train Loss: 0.11399 Vali.MSE: 0.40856 Vali.MAE: 0.27501 Test.MSE: 0.16352 Test.MAE: 0.20545
	EarlyStopping counter: 2 out of 5
Updating learning rate to 0.006434312
Epoch 4: cost time: 47.96 sec
	Epoch 4: Steps- 286 | Train Loss: 0.11351 Vali.MSE: 0.40447 Vali.MAE: 0.27015 Test.MSE: 0.16396 Test.MAE: 0.20548
	EarlyStopping counter: 3 out of 5
Updating learning rate to 0.0057908808
Epoch 5: cost time: 49.31 sec
	Epoch 5: Steps- 286 | Train Loss: 0.11239 Vali.MSE: 0.40338 Vali.MAE: 0.26648 Test.MSE: 0.16419 Test.MAE: 0.20461
	EarlyStopping counter: 4 out of 5
Updating learning rate to 0.00521179272
Epoch 6: cost time: 45.87 sec
	Epoch 6: Steps- 286 | Train Loss: 0.11197 Vali.MSE: 0.40589 Vali.MAE: 0.27102 Test.MSE: 0.16333 Test.MAE: 0.20529
	EarlyStopping counter: 5 out of 5
	Early stopping
Start Testing- WPMixer_Weather_dec-True_sl96_pl96_dm128_bt128_wvsym3_tf3_df5_ptl16_stl8_sd42
test 10444
mse: 0.16451449692249298, mae: 0.2073848843574524

**Comment by MNmurad** [comment]
You ran the script for the unified results (Table 9). As I can see, you got almost the same results in your re-evaluation. Please check the README file. If you want to check the results of Table-2, use - ```bash ./scripts/Full_HyperSearch/Weather_full_hyp.sh```

**Comment by MNmurad** [comment]
We have two experimental settings:
1. Table-9: Unified: Look-back window length is 96.
2. Table-2: Hyperparam search: we search for the optimum look-back window value.
